homeassistant/components/resolution_center/issue_registry.py

```diff
@@ -0,0 +1,140 @@
+"""Persistently store issues raised by integrations."""
+from __future__ import annotations
+
+import dataclasses
+from typing import cast
+
+from homeassistant.const import MAJOR_VERSION, MINOR_VERSION, PATCH_VERSION
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.storage import Store
+
+DATA_REGISTRY = "issue_registry"
+STORAGE_KEY = "resolution_center.issue_registry"
+STORAGE_VERSION = 1
+SAVE_DELAY = 10
+
+
+@dataclasses.dataclass(frozen=True)
+class IssueEntry:
+    """Issue Registry Entry."""
+
+    dismissed_version_major: int | None
+    dismissed_version_minor: int | None
+    dismissed_version_patch: int | None
```

Comment:
We also have dev/beta/nightly markers, how would those work?

Reply:
Those are just patch versions, aren't they?

Reply:
Not really? Pre-release or build annotations are additional and actually often have meaning. We take them into account in Home Assistant (e.g., we can actually compare `2022.7.0b0 to be smaller than 2022.7.0 using AwesomeVersion)

<https://github.com/ludeeus/awesomeversion/blob/main/awesomeversion/awesomeversion.py#L339>

With the current structure, dismissing 2022.7.0b1, would also dismiss 2022.7.0

As an alternative approach, we could also just store the version it was dismissed at (raw, version, as string). We can always parse it again quickly with awesome version and get the major, minor, patch and modifier from it again.